jf_agent/main.py

```diff
@@ -366,37 +370,53 @@ def download_data(config, creds, endpoint_jira_info, endpoint_git_instances_info
             print_all_jira_fields(config, jira_connection)
         download_data_status.append(load_and_dump_jira(config, endpoint_jira_info, jira_connection))
 
-    is_multi_git_config = len(config.git_configs) > 1
-    for git_config in config.git_configs:
-        agent_logging.log_and_print(
-            logger,
-            logging.INFO,
-            f'Obtained {git_config.git_provider} configuration, attempting download...',
-        )
-        if is_multi_git_config:
-            instance_slug = git_config.git_instance_slug
-            instance_info = endpoint_git_instances_info.get(instance_slug)
-            instance_creds = creds.git_instance_to_creds.get(instance_slug)
-        else:
-            # support legacy single-git support, which assumes only one available git instance
-            instance_info = list(endpoint_git_instances_info.values())[0]
-            instance_creds = list(creds.git_instance_to_creds.values())[0]
+    
+    # git downloading is parallelized by the number of configurations. 
+    futures = []
+    with ThreadPoolExecutor(max_workers=8) as executor:
```

Comment:
I think we should probably make max_workers be parametrizable, with the ability to override (set it to something smaller than the default) it in the config file. With the default being the min of (number of instances defined in the config file) or (something non-crazy like 8?).

Reply:
https://github.com/Jellyfish-AI/jf_agent/pull/201/commits/ef02b7c0d3264832cc3fe67037efc4754bc60344